### PR TITLE
Feature/games 1072/bitcoinrpc

### PR DESCRIPTION
**Pull request by Extrodox**
### What does this PR do?  
_Implement the RPC Adapter class._

### Link to Jira

[GAMES-1072](https://bitcoindotcom.atlassian.net/browse/GAMES-1072)

### Steps for testing  
1. Call any method using BTC or BCH with RPC Adapter.

### Deployment notes
_Please be sure to describe any special deployment notes for this change, including database changes, new environment variables, client updates, etc._

- [ ] Database migration required.  Script: ______

### CODE REVIEW

File: wallet_lib/adapters/rpc_Adapter.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    def __init__(self, result, error=None, code=None):
+        self.result = result
+        self.error = error
+        self.code = code
+
+
+class RPCAdapter(WalletAdapterBase):
+    def __init__(self, rpc_user, rpc_password, rpc_url, rpc_port):
+        self.log = logging.getLogger('RPCCaller')
```
Comment (fabasoad) [inline wallet_lib/adapters/rpc_Adapter.py]:
```suggestion
        self.log = logging.getLogger('RPCAdapter')
```

File: wallet_lib/wallet_factory.py [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+    _SUPPORTED_WALLETS = {
+        'BCH': BCHWallet,
+        'BTC': BTCWallet
+    }
+
+    def __init__(self):
+        pass
 
-    def get(self, ticker_symbol):
+    def get(self, ticker_symbol, adapter: WalletAdapterBase):
```
Comment (fabasoad) [inline wallet_lib/wallet_factory.py]:
```suggestion
    def get(self, ticker_symbol, adapter: WalletAdapterBase = RPCAdapter(..args)):
```
Would be good to have a default strategy, like RPCAdapter. So the lib consumer should not always define it.
I propose to get RPC connection properties from environment variables.

**Review by fabasoad — CHANGES_REQUESTED** [review]
Also, could you please merge your unit tests into this branch as well. Pros:
1. You will be able to merge PR as for now it is blocked by CI.
2. We can deliver feature+tests by one merging and without separating them into multiple deliveries.

File: wallet_lib/adapters/rpc_Adapter.py (hunk 1 above)
Comment (Extrodox) [reply wallet_lib/adapters/rpc_Adapter.py]: This logger naming issue fixes.

**Review by fabasoad — CHANGES_REQUESTED** [review]
1. CI is failing and basically locally tests are failing as well.
2. Regarding tests - I see that right now we have: bch_wallet, btc_wallet, rpc_bch_wallet, rpc_btc_wallet, rpc_wallet tests. It is not really clear what do they do. What test should we add when we will add zec wallet and dash? We should test each class separately. In another words file structure should be the same as in main source folder but with `_test` suffix. Also, it means that each test responsible for specific logical unit, for specific class. For example, bch_wallet_test should test bch wallet only without dependency on specific adapter implementation, because specific adapter implementation should be tested in their own test classes like rpc_adapter_test and cmd_adapter_test. When we will add new wallet, let's say zec wallet - all what we need to do is just to add one test class to test zec wallet itself and we will not need to test rpc adapter or any other adapter in scope of it. Please feel free to contact me directly in case you have any questions.

_How tests folder should look like:_
```bash
wallet_lib\tests
│   bch_wallet_test.py
│   btc_wallet_test.py
│   dash_wallet_test.py
│   wallet_exceptions_test.py
│   wallet_factory_test.py
│   zec_wallet_test.py
│
└───adapters
        cmd_adapter_test.py
        rpc_adapter_test.py
```
3. Please rename _rpc_Adapter.py_ to _rpc_adapter.py_.
4. I have tried to use the wallet_lib by connecting it with my temp project but have an error. Here is the screenshot:

Annotation 2019-12-12 085741

**Comment by Extrodox** [comment]
@fabasoad the file is already a renamed. Can you please check on you side.
May be this is windows thing. 
Screenshot 2019-12-12 at 7 06 10 AM

**Comment by Extrodox** [comment]
Screenshot 2019-12-12 at 7 10 07 AM

**Review by fabasoad — CHANGES_REQUESTED** [review]
Discussed in Slack

**Comment by Extrodox** [comment]
I have not created this wallet_exceptions_test.py file except that those exceptions are already handled in the wallet_factory_test.py file.

**Review by fabasoad — APPROVED** [review]
Looks good!
I had an issue when running tests locally because of missing env variables. I've added fix for that. Also, I've added test coverage plugin for CI to keep tests healthy. For now tests coverage report looks like this:
```bash
----------- coverage: platform win32, python 3.8.0-final-0 -----------
Name                                         Stmts   Miss  Cover
----------------------------------------------------------------
wallet_lib\__init__.py                           5      0   100%
wallet_lib\__main__.py                           0      0   100%
wallet_lib\adapters\__init__.py                  3      0   100%
wallet_lib\adapters\cmd_adapter.py              26      0   100%
wallet_lib\adapters\rpc_adapter.py              35      1    97%
wallet_lib\adapters\wallet_adapter_base.py       3      1    67%
wallet_lib\bch_wallet.py                        55     35    36%
wallet_lib\btc_wallet.py                        53     33    38%
wallet_lib\dash_wallet.py                        2      0   100%
wallet_lib\wallet_exceptions.py                  6      0   100%
wallet_lib\wallet_factory.py                    15      0   100%
wallet_lib\zec_wallet.py                         2      0   100%
----------------------------------------------------------------
TOTAL                                          205     70    66%

Required test coverage of 65.85% reached. Total coverage: 65.85%
```
Not perfect but not bad (65.85% is very good result). Let's keep it as a minimum starting from now.
Please take a look at commit that I've pushed and feel free to merge PR in case you're fine with that.

**Pull request merged**